### PR TITLE
[SIW-755] Update `react-native` to version `0.70.14`

### PR DESCRIPTION
**Pull request by hevelius**
## Short description
This PR updates react-native to resolve an error related to the new version of xcode. ref: https://github.com/facebook/react-native/issues/34106

## List of changes proposed in this pull request
- Update package.json
- Update pod

## How to test
Run the example app. It should works correctly

### CODE REVIEW

File: example/.ruby-version [hunk 1]
```diff
@@ -1 +0,0 @@
-2.7.5
```
Comment (shadowsheep1) [inline example/.ruby-version:1]: Why did we remove this?

File: example/Gemfile [hunk 2]
```diff
@@ -1,6 +1,7 @@
 source 'https://rubygems.org'
 
 # You may use http://rbenv.org/ or https://rvm.io/ to install and use this version
-ruby '2.7.5'
+ruby '>= 2.6.10'
```
Comment (shadowsheep1) [inline example/Gemfile:4]: If the ruby version needed by RN `0.70.14` is greater than or equal to `2.6.10`, why do not leave the actual `2.7.5` version?

**Review by shadowsheep1 — CHANGES_REQUESTED** [review]

File: example/.ruby-version (hunk 1 above)
Comment (hevelius) [reply example/.ruby-version]: Following a migration tool to update RN at `0.70.14` the .ruby-version was marked as removed. I have restore this file here https://github.com/pagopa/io-react-native-crypto/pull/27/commits/5ef0dac32617817fda0e612e7b670701ef8727f5

File: example/Gemfile (hunk 2 above)
Comment (hevelius) [reply example/Gemfile]: Yep Copy/Paste error. Good catch 🌈 Thx. Fixed here https://github.com/pagopa/io-react-native-crypto/pull/27/commits/5ef0dac32617817fda0e612e7b670701ef8727f5

**Review by shadowsheep1 — APPROVED** [review]
LGTM

**Pull request merged**